src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -259,21 +260,35 @@
     (= (:type x)
        :puppetlabs.services.request-handler.request-handler-core/bad-request)))
 
+(defn jruby-timeout?
+  [x]
+  "Determine if the supplied slingshot message is for a JRuby borrow timeout."
+  (when (map? x)
+    (= (:type x)
+       ::jruby/jruby-timeout)))
```

Comment:
Yup, we learned that `::foo/bar` does a lookup for a namespace alias for `foo`, which is why `jruby` was working in that case.  Which actually seems like a reasonable syntax to me.